builder/parser/line_parsers.go

```diff
@@ -233,7 +233,7 @@ func parseString(rest string) (*Node, map[string]bool, error) {
 // parseJSON converts JSON arrays to an AST.
 func parseJSON(rest string) (*Node, map[string]bool, error) {
 	var myJson []interface{}
-	if err := json.Unmarshal([]byte(rest), &myJson); err != nil {
+	if err := json.NewDecoder(strings.NewReader(rest)).Decode(&myJson); err != nil {
```

Comment:
Actually it isn't an issue. `json.Decode` unmarshal whole content from `Reader` anyway if we have only one object(90% of cases in docker) and it is same by memory consumption. But `Decoder` uses internal `sync.Pool` for storing some data structures in cache.